### ISSUE TITLE
Exception: Must be instance of `Stream`

### ISSUE THREAD

**Issue by njam** [cargomedia/cm-janus#143]
@tomaszdurka @kris-lab @vogdb 

```
app info {"code":490,"message":"Cannot subscribe: null error: Must be instance of `Stream`"}
app debug -> browser {"janus":"error","error":{"code":490,"reason":"Cannot subscribe: null error: Must be instance of `Stream`"}}
```

**Comment by tomaszdurka** [comment]
Fixed by #122.

**Comment by vogdb** [comment]
hm. What a magical #122.

**Comment by tomaszdurka** [comment]
^^

**Comment by kris-lab** [comment]
```
app info cm-api request https://alicex.dev.cargomedia.ch/rpc/null {"method":"CM_Janus_RpcEndpoints.removeStream","params":["mad-panda","k.5gCAnFWajejoyccAuLvw__","fb00fe1a-bf4f-4c14-bf9c-e23f498b5b51"]}
app info Removed Plugin{"id":3276553429,"type":"janus.plugin.cm.audioroom"}
app debug -> browser {"janus":"hangup","session_id":1798298769,"sender":3276553429,"reason":"DTLS alert"}
app debug <- janus {"janus":"detached","session_id":1798298769,"sender":3276553429}
app debug <- janus {"janus":"success","session_id":1798298769,"transaction":"UpC77NgdHfEH"}
app debug <- janus {"janus":"detached","session_id":1169301656,"sender":1583884922}
app debug <- janus {"janus":"success","session_id":1169301656,"transaction":"F8OACSI3hevf"}
app debug -> browser {"janus":"detached","session_id":1798298769,"sender":3276553429}
app debug -> browser {"janus":"success","session_id":1798298769,"transaction":"UpC77NgdHfEH"}
app debug -> browser {"janus":"detached","session_id":1169301656,"sender":1583884922}
app debug -> browser {"janus":"success","session_id":1169301656,"transaction":"F8OACSI3hevf"}
app error {"message":"No handler found for `janus.plugin.cm.audioroom`, `archive-finished`"}
app info cm-api response 
app error Error: Must be instance of `Stream`
    at Streams.add (/usr/lib/node_modules/cm-janus/lib/streams.js:14:11)
    at /usr/lib/node_modules/cm-janus/lib/janus/plugin/streaming.js:34:37
    at tryCatcher (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

**Comment by tomaszdurka** [comment]
Seems it's not so magical :)

**Comment by njam** [comment]
```
app debug <- janus {"janus":"webrtcup","session_id":4260574594,"sender":2622110773}
app info cm-api request https://alicex.dev.cargomedia.ch/rpc/null {"method":"CM_Janus_RpcEndpoints.subscribe","params":["mad-panda","{\"sessionId\":\"6502c8d7804ea47642d210e63eb565f8\"}","G2YkvQ2jsAShFsV5R7R4QA__","9ae664a191ba6693baa2c477e52844fd","{\"streamChannelType\":270}","991834b0-ff40-4146-b088-9cd4033c3a44",1453202206.795]}
app debug <- janus {"janus":"hangup","session_id":4260574594,"sender":2622110773,"reason":"DTLS alert"}
app info cm-api request https://alicex.dev.cargomedia.ch/rpc/null {"method":"CM_Janus_RpcEndpoints.removeStream","params":["mad-panda","G2YkvQ2jsAShFsV5R7R4QA__","991834b0-ff40-4146-b088-9cd4033c3a44"]}
app info Removed Plugin{"id":2622110773,"type":"janus.plugin.cm.audioroom"}
app debug -> browser {"janus":"event","session_id":1562003312,"sender":378349530,"plugindata":{"plugin":"janus.plugin.cm.audioroom","data":{"audioroom":"event","id":"G2YkvQ2jsAShFsV5R7R4QA__","leaving":2467807166}}}
app debug -> browser {"janus":"hangup","session_id":4260574594,"sender":2622110773,"reason":"DTLS alert"}
app debug <- janus {"janus":"detached","session_id":4260574594,"sender":2622110773}
app debug <- janus {"janus":"success","session_id":4260574594,"transaction":"ZdAK3gcLBhvy"}
app debug -> browser {"janus":"detached","session_id":4260574594,"sender":2622110773}
app debug -> browser {"janus":"success","session_id":4260574594,"transaction":"ZdAK3gcLBhvy"}
app info cm-api response 
app info cm-api response 
app error Error: Must be instance of `Stream`
    at Streams.add (/usr/lib/node_modules/cm-janus/lib/streams.js:14:11)
    at /usr/lib/node_modules/cm-janus/lib/janus/plugin/streaming.js:34:37
    at tryCatcher (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

**Comment by tomaszdurka** [comment]
Fixed in #160 

`.stream` access has been replaced with `getStream()` method which will early verify that plugin has valid stream otherwise throw Janus exception properly handled by proxy.
